src/StreamJsonRpc.Shared/JsonRpc.cs

```diff
@@ -663,48 +663,79 @@ protected virtual Task<ReturnType> InvokeCoreAsync<ReturnType>(int? id, string t
+                                throw new ArgumentException(string.Format(CultureInfo.CurrentCulture, Resources.OverloadsMissingMethodAttribute, method.Name, nameof(JsonRpcMethodAttribute)));
+                            }
+
+                            string attributeValue;
+                            if (clrMethodToRequestMethodMap.TryGetValue(method.Name, out attributeValue))
+                            {
+                                if (!string.Equals(attribute.Name, attributeValue, StringComparison.Ordinal))
+                                {
+                                    throw new ArgumentException(string.Format(CultureInfo.CurrentCulture, Resources.ConflictingMethodNameAttribute, method.Name, nameof(JsonRpcMethodAttribute)));
+                                }
+                            }
+                            else
                             {
-                                if (!string.Equals(attribute.Name, value, StringComparison.Ordinal))
+                                clrMethodToRequestMethodMap.Add(method.Name, attribute.Name);
+                            }
+
+                            string methodValue;
+                            if (requestMethodToClrMethodMap.TryGetValue(attribute.Name, out methodValue))
+                            {
+                                if (!string.Equals(method.Name, methodValue, StringComparison.Ordinal))
                                 {
-                                    throw new RemoteTargetException(string.Format(CultureInfo.CurrentCulture, Resources.ConflictingMethodNameAttribute, method.Name, nameof(JsonRpcMethodAttribute)));
+                                    throw new ArgumentException(string.Format(CultureInfo.CurrentCulture, Resources.ConflictingMethodAttributeValue, nameof(JsonRpcMethodAttribute), attribute.Name));
                                 }
                             }
                             else
                             {
-                                methodToAttributeMap.Add(method.Name, attribute.Name);
-                            }                            
+                                requestMethodToClrMethodMap.Add(attribute.Name, method.Name);
+                            }
                         }
                     }
                 }
             }
 
-            // If not conflicts are find, then create another map which reverses the key/value mapping.
-            var attributeToMethodMap = new Dictionary<string, string>(StringComparer.Ordinal);
-            foreach (var method in methodToAttributeMap)
+            // We shouldn't ever hit this unless there's a bug in the above code.  Put in so we can detect bugs easier.
+            if (clrMethodToRequestMethodMap.Count != requestMethodToClrMethodMap.Count)
             {
-                attributeToMethodMap.Add(method.Value, method.Key);
+                throw new InvalidOperationException("There was an error reflecting over the target object.");
```

Comment:
Please replace this if block with this:
```csharp
Assumes.True(clrMethodToRequestMethodMap.Count == requestMethodToClrMethodMap.Count, "There was an error reflecting over the target object.");
```

That throws an internal error exception, which is more appropriate given your comment than an InvalidOperationException.